core/src/main/java/hudson/util/ProcessTree.java

```diff
@@ -1155,25 +1095,88 @@ private AIXProcess(int pid) throws IOException {
                     //   uint64_t pr__pad[8];             /* reserved for future use */
                     //   struct   lwpsinfo pr_lwp;        /* "representative" thread info */
                     // }
-
                     psinfo.seek(48); // offset of pr_pid
 
                     if (adjust((int)psinfo.readLong()) != pid)
                         throw new IOException("psinfo PID mismatch"); // sanity check
 
-                    if (adjust((int)psinfo.readLong()) != ppid)
-                        throw new IOException("psinfo PPID mismatch"); // sanity check
+                    ppid = adjust((int)psinfo.readLong()); // AIX pids are stored as a 64 bit integer, 
+                                                           // but the first 4 bytes are always 0
 
                     psinfo.seek(148); // offset of pr_argc
 
                     argc = adjust(psinfo.readInt());
                     pr_argp = adjustL(psinfo.readLong());
                     pr_envp = adjustL(psinfo.readLong());
+
+                    b64 = is64BitProcess();
                 } finally {
                     psinfo.close();
                 }
             }
 
+            protected boolean is64BitProcess() {
```

Comment:
Do you plan to have any API consumers outside? If not, it might be better to just make it private